packages/commerce-sdk-react/src/hooks/ShopperBaskets/mutation.ts

```diff
@@ -333,43 +335,106 @@ the body are the following properties if specified:
 
+export const getQueryKeysMatrix = (customerId: string | null) => {
+    return {
+        addCouponToBasket: (
+            params: Argument<Client['addCouponToBasket']>,
+            response: DataType<Client['addCouponToBasket']>
+        ): QueryKeysMatrixElement => {
+            const basketId = params.parameters.basketId
+
+            return {
+                update: [['/baskets', basketId, {basketId}]],
+                ...(customerId ? {invalidate: [['/customers', customerId, '/baskets']]} : {})
+            }
+        },
+        addItemToBasket: (
+            params: Argument<Client['addItemToBasket']>,
+            response: DataType<Client['addItemToBasket']>
+        ): QueryKeysMatrixElement => {
+            const basketId = params.parameters.basketId
+
+            return {
+                update: [['/baskets', basketId, {basketId}]],
+                ...(customerId ? {invalidate: [['/customers', customerId, '/baskets']]} : {})
+            }
+        },
+        addPaymentInstrumentToBasket: (
+            params: Argument<Client['addPaymentInstrumentToBasket']>,
+            response: DataType<Client['addPaymentInstrumentToBasket']>
+        ): QueryKeysMatrixElement => {
+            const basketId = params.parameters.basketId
+
+            return {
+                update: [['/baskets', basketId, {basketId}]],
+                ...(customerId ? {invalidate: [['/customers', customerId, '/baskets']]} : {})
+            }
+        },
+        createBasket: (
+            params: Argument<Client['createBasket']>,
+            response: DataType<Client['createBasket']>
+        ): QueryKeysMatrixElement => {
+            const basketId = response.basketId
+
+            return {
+                ...(basketId ? {update: [['/baskets', basketId, {basketId}]]} : {}),
+                ...(customerId ? {invalidate: [['/customers', customerId, '/baskets']]} : {})
+            }
+        },
+        deleteBasket: (
+            params: Argument<Client['deleteBasket']>,
+            response: DataType<Client['deleteBasket']>
+        ): QueryKeysMatrixElement => {
+            const basketId = params?.parameters.basketId
+
+            return {
+                ...(customerId ? {invalidate: [['/customers', customerId, '/baskets']]} : {}),
+                ...(basketId ? {remove: [['/baskets', basketId]]} : {})
+            }
+        },
+        // TODO: test in test-commerce project (http://localhost:3000/basket)
+        updateBasket: (
+            params: Argument<Client['updateBasket']>,
+            response: DataType<Client['updateBasket']>
+        ): QueryKeysMatrixElement => {
+            const basketId = response.basketId
+
+            return {
+                // NOTE: this is the query key for useBasket hook
+                // TODO: but we're missing headers, rawResponse -> not only {basketId}
+                ...(basketId ? {update: [['/baskets', basketId, {basketId}]]} : {}),
```

Comment:
@adamraya i think we discussed this issue for those cache updates and we thought this is not a problem, but i couldn't remember why...

How do we handle `headers` and `rawResponses` for cache updates?

Reply:
To be very clear, I believe cache update and invalidate is different.

We don't need `headers` or `rawResponses` or any other options for invalidations because the fuzzy matching algo provided by react query.

So imagine you have a cache entry that looks like:

```
{
   ['/product', 123, {headers:'myheaders'}]
}
```

if you call `queryClient.invalidateQueries(['/product', 123])`, that cache entry will get invalidated.

TL;DR this is not a problem for cache invalidation, but i'm not sure about cache updates.

Reply:
Unlike invalidation/remove, cache updates do not use fuzzy matching.. they're very exact.

Reply:
The React Query `setQueryData` API we use in cache updates use strict matching. If there's no match a new entry in cache is created.

The React Query `invalidateQueries` and `removeQueries` APIs does a partial/fuzzy query matching only when we omit the `arg` object in the query key declared in the queryKeyMatrix.
If the query key declared inside the queryKeyMatrix includes the `arg` object, we do a strict matching of the stringified version of the `arg` object.

We probably need reconsider including the `arg` object in the query key structure.